src/pages/portfolio/index.tsx

```diff
@@ -1,11 +1,49 @@
 'use client';
 
+import React from 'react';
+import { XCircle } from 'lucide-react';
+import { Button } from '@penumbra-zone/ui/Button';
 import { Text } from '@penumbra-zone/ui/Text';
+import { Density } from '@penumbra-zone/ui/Density';
+import mobile from 'is-mobile';
 
 export const PortfolioPage = () => {
-  return (
-    <section>
-      <Text h2>Hi!</Text>
+  const isOnMobile = mobile();
```

Comment:
issue (blocking): this `mobile()` call results in the hydration failure because this code doesn't run on the server. To render it correctly on the server, the library proposes passing the nextjs' `req` object as the function argument.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/3ee12947-8e49-44f3-acdb-3685d4841330">

Reply:
We can't get the req object easily because we are using /pages, not /app. Using useEffect to get around this.

Reply:
Maybe using the Nextjs' `userAgent` function in the `middleware` might be a solution? https://nextjs.org/docs/app/api-reference/functions/userAgent

Reply:
I don't believe this would help us - this augments the request, but we still can't get the req object, without modifying the structure of the app. Not worth it imo, useEffect is just fine.

Reply:
Fixed this, really happy with the result now.

Reply:
LMK if it works for you.